## EXPLORATORY DATA ANALYSIS OF AMCAT DATA USING PYTHON

## Exploratory Data Analysis (EDA), is essentially a type of story telling for statisticians. It allows us to uncover patterns and insights, often with visual methods, within data.

## STEP - I: IMPORTING THE NECESSARY LIBRARIES

In [1]:
import numpy as np
import pandas as pd

## NUMPY: It's a Python library used for working with arrays, also has functions for working in domain of linear algebra, fourier transform, and matrices. NumPy was created in 2005 by Travis Oliphant. It is an open source project and you can use it freely. NumPy stands for Numerical Python.

## PANDAS: It's a software library written for the Python programming language for data manipulation and analysis. In particular, it offers data structures and operations for manipulating numerical tables and time series.

## STEP - II: IMPORTING THE DATA 

In [2]:
df = pd.read_excel(r'C:\Users\admin\Desktop\KANAV BANSAL\KANAV TASKS\TASK - II - EDA - AMCAT DATA/reports.xlsx')
df

Unnamed: 0      ID   Salary        DOJ                  DOL  \
0         train  203097   420000 2012-06-01              present   
1         train  579905   500000 2013-09-01              present   
2         train  810601   325000 2014-06-01              present   
3         train  267447  1100000 2011-07-01              present   
4         train  343523   200000 2014-03-01  2015-03-01 00:00:00   
...         ...     ...      ...        ...                  ...   
3993      train   47916   280000 2011-10-01  2012-10-01 00:00:00   
3994      train  752781   100000 2013-07-01  2013-07-01 00:00:00   
3995      train  355888   320000 2013-07-01              present   
3996      train  947111   200000 2014-07-01  2015-01-01 00:00:00   
3997      train  324966   400000 2013-02-01              present   

                      Designation           JobCity Gender        DOB  \
0         senior quality engineer         Bangalore      f 1990-02-19   
1               assistant manager            Indore      m 1989-10-04   
2                systems engineer           Chennai      f 1992-08-03   
3        senior software engineer           Gurgaon      m 1989-12-05   
4                             get           Manesar      m 1991-02-27   
...                           ...               ...    ...        ...   
3993            software engineer        New Delhi       m 1987-04-15   
3994             technical writer         Hyderabad      f 1992-08-27   
3995  associate software engineer         Bangalore      m 1991-07-03   
3996           software developer  Asifabadbanglore      f 1992-03-20   
3997      senior systems engineer           Chennai      f 1991-02-26   

      10percentage  ... ComputerScience  MechanicalEngg  ElectricalEngg  \
0            84.30  ...              -1              -1              -1   
1            85.40  ...              -1              -1              -1   
2            85.00  ...              -1              -1              -1   
3            85.60  ...              -1              -1              -1   
4            78.00  ...              -1              -1              -1   
...            ...  ...             ...             ...             ...   
3993         52.09  ...              -1              -1              -1   
3994         90.00  ...              -1              -1              -1   
3995         81.86  ...              -1              -1              -1   
3996         78.72  ...             438              -1              -1   
3997         70.60  ...              -1              -1              -1   

     TelecomEngg  CivilEngg  conscientiousness agreeableness extraversion  \
0             -1         -1             0.9737        0.8128       0.5269   
1             -1         -1            -0.7335        0.3789       1.2396   
2             -1         -1             0.2718        1.7109       0.1637   
3             -1         -1             0.0464        0.3448      -0.3440   
4             -1         -1            -0.8810       -0.2793      -1.0697   
...          ...        ...                ...           ...          ...   
3993          -1         -1            -0.1082        0.3448       0.2366   
3994          -1         -1            -0.3027        0.8784       0.9322   
3995          -1         -1            -1.5765       -1.5273      -1.5051   
3996          -1         -1            -0.1590        0.0459      -0.4511   
3997          -1         -1            -1.1128       -0.2793      -0.6343   

      nueroticism  openess_to_experience  
0         1.35490                -0.4455  
1        -0.10760                 0.8637  
2        -0.86820                 0.6721  
3        -0.40780                -0.9194  
4         0.09163                -0.1295  
...           ...                    ...  
3993      0.64980                -0.9194  
3994      0.77980                -0.0943  
3995     -1.31840                -0.7615  
3996     -0.36120                -0.0943  
3997      1.

## Observation :
There are 3998 data of the people which contains the Gender, Designation, Salary, Location, Percentage of 10th, 12th and B.Tech including with College Tier -Specialization, etc things.

## STEP - III: DATA ANALYSIS

## include = 'all' is provided as an option, the result will include a union of attributes of each type. The includes and excludes parameters, can be used to limit which columns in a DataFrame are analyzed for the output. The parameters are ignored when analyzing a Series & Describes a numeric Series.

In [3]:
df.describe(include = 'all')

Unnamed: 0            ID        Salary                  DOJ      DOL  \
count        3998  3.998000e+03  3.998000e+03                 3998     3998   
unique          1           NaN           NaN                   81       67   
top         train           NaN           NaN  2014-07-01 00:00:00  present   
freq         3998           NaN           NaN                  199     1875   
first         NaN           NaN           NaN  1991-06-01 00:00:00      NaN   
last          NaN           NaN           NaN  2015-12-01 00:00:00      NaN   
mean          NaN  6.637945e+05  3.076998e+05                  NaN      NaN   
std           NaN  3.632182e+05  2.127375e+05                  NaN      NaN   
min           NaN  1.124400e+04  3.500000e+04                  NaN      NaN   
25%           NaN  3.342842e+05  1.800000e+05                  NaN      NaN   
50%           NaN  6.396000e+05  3.000000e+05                  NaN      NaN   
75%           NaN  9.904800e+05  3.700000e+05                  NaN      NaN   
max           NaN  1.298275e+06  4.000000e+06                  NaN      NaN   

              Designation    JobCity Gender                  DOB  \
count                3998       3998   3998                 3998   
unique                419        339      2                 1872   
top     software engineer  Bangalore      m  1991-01-01 00:00:00   
freq                  539        627   3041                   11   
first                 NaN        NaN    NaN  1977-10-30 00:00:00   
last                  NaN        NaN    NaN  1997-05-27 00:00:00   
mean                  NaN        NaN    NaN                  NaN   
std                   NaN        NaN    NaN                  NaN   
min                   NaN        NaN    NaN                  NaN   
25%                   NaN        NaN    NaN                  NaN   
50%                   NaN        NaN    NaN                  NaN   
75%                   NaN        NaN    NaN                  NaN   
max                   NaN        NaN    NaN                  NaN   

        10percentage  ... ComputerScience  MechanicalEngg  ElectricalEngg  \
count    3998.000000  ...     3998.000000     3998.000000     3998.000000   
unique           NaN  ...             NaN             NaN             NaN   
top              NaN  ...             NaN             NaN             NaN   
freq             NaN  ...             NaN             NaN             NaN   
first            NaN  ...             NaN             NaN             NaN   
last             NaN  ...             NaN             NaN             NaN   
mean       77.925443  ...       90.742371       22.974737       16.478739   
std         9.850162  ...      175.273083       98.123311       87.585634   
min        43.000000  ...       -1.000000       -1.000000       -1.000000   
25%        71.680000  ...       -1.000000       -1.000000       -1.000000   
50%        79.150000  ...       -1.000000       -1.000000       -1.000000   
75%        85.670000  ...       -1.000000       -1.000000       -1.000000   
max        97.760000  ...      715.000000      623.000000      676.000000   

        TelecomEngg    CivilEngg  conscientiousness agreeableness  \
count   3998.000000  3998.000000        3998.000000   3998.000000   
unique          NaN          NaN                NaN           NaN   
top             NaN          NaN                NaN           NaN   
freq            NaN          NaN                NaN           NaN   
first           NaN          NaN                NaN           NaN   
last            NaN          NaN                NaN           NaN   
mean      31.851176     2.683842          -0.037831      0.146496   
std      104.852845    36.658505           1.028666      0.941782   
min       -1.000000    -1.000000          -4.126700     -5.781600   
25%       -1.000000    -1.000000          -0.713525     -0.287100   
50%       -1.000000    -1.000000           0.046400      0.212400   
75%       -1.000000    -1.000000        

## Obesrvation:
We are able to get the information of MEAN, MEDIAN, STANDARD DEVIATION, MINIMUM & MAXIMUM VALUES OF ENTIRE DATA

## GETTING THE TOP 5 DATA

In [4]:
X = df.head()
X

Unnamed: 0      ID   Salary        DOJ                  DOL  \
0      train  203097   420000 2012-06-01              present   
1      train  579905   500000 2013-09-01              present   
2      train  810601   325000 2014-06-01              present   
3      train  267447  1100000 2011-07-01              present   
4      train  343523   200000 2014-03-01  2015-03-01 00:00:00   

                Designation    JobCity Gender        DOB  10percentage  ...  \
0   senior quality engineer  Bangalore      f 1990-02-19          84.3  ...   
1         assistant manager     Indore      m 1989-10-04          85.4  ...   
2          systems engineer    Chennai      f 1992-08-03          85.0  ...   
3  senior software engineer    Gurgaon      m 1989-12-05          85.6  ...   
4                       get    Manesar      m 1991-02-27          78.0  ...   

  ComputerScience  MechanicalEngg  ElectricalEngg TelecomEngg  CivilEngg  \
0              -1              -1              -1          -1         -1   
1              -1              -1              -1          -1         -1   
2              -1              -1              -1          -1         -1   
3              -1              -1              -1          -1         -1   
4              -1              -1              -1          -1         -1   

   conscientiousness agreeableness extraversion  nueroticism  \
0             0.9737        0.8128       0.5269      1.35490   
1            -0.7335        0.3789       1.2396     -0.10760   
2             0.2718        1.7109       0.1637     -0.86820   
3             0.0464        0.3448      -0.3440     -0.40780   
4            -0.8810       -0.2793      -1.0697      0.09163   

   openess_to_experience  
0                -0.4455  
1                 0.8637  
2                 0.6721  
3                -0.9194  
4                -0.1295  

[5 rows x 39 columns]

## OBSERVATION:
There are 3 males & 2 females, where one of the male has highest Salary Package who is a SENIOR SOFTWARE ENGINEER.

## GETTING THE LAST 5 DATA

In [5]:
Y = df.tail()
Y

Unnamed: 0      ID  Salary        DOJ                  DOL  \
3993      train   47916  280000 2011-10-01  2012-10-01 00:00:00   
3994      train  752781  100000 2013-07-01  2013-07-01 00:00:00   
3995      train  355888  320000 2013-07-01              present   
3996      train  947111  200000 2014-07-01  2015-01-01 00:00:00   
3997      train  324966  400000 2013-02-01              present   

                      Designation           JobCity Gender        DOB  \
3993            software engineer        New Delhi       m 1987-04-15   
3994             technical writer         Hyderabad      f 1992-08-27   
3995  associate software engineer         Bangalore      m 1991-07-03   
3996           software developer  Asifabadbanglore      f 1992-03-20   
3997      senior systems engineer           Chennai      f 1991-02-26   

      10percentage  ... ComputerScience  MechanicalEngg  ElectricalEngg  \
3993         52.09  ...              -1              -1              -1   
3994         90.00  ...              -1              -1              -1   
3995         81.86  ...              -1              -1              -1   
3996         78.72  ...             438              -1              -1   
3997         70.60  ...              -1              -1              -1   

     TelecomEngg  CivilEngg  conscientiousness agreeableness extraversion  \
3993          -1         -1            -0.1082        0.3448       0.2366   
3994          -1         -1            -0.3027        0.8784       0.9322   
3995          -1         -1            -1.5765       -1.5273      -1.5051   
3996          -1         -1            -0.1590        0.0459      -0.4511   
3997          -1         -1            -1.1128       -0.2793      -0.6343   

      nueroticism  openess_to_experience  
3993      0.64980                -0.9194  
3994      0.77980                -0.0943  
3995     -1.31840                -0.7615  
3996     -0.36120                -0.0943  
3997      1.32553                -0.6035  

[5 rows x 39 columns]

## Observation:
There are 3 females & 2 males, where one of the female has highest Salary Package who is a SENIOR SYSTEMS ENGINEER.

## value_counts( ): Return a Series containing counts of unique rows in the DataFrame.

In [6]:
df.Designation.value_counts()

software engineer                 539
software developer                265
system engineer                   205
programmer analyst                139
systems engineer                  118
                                 ... 
associate technical operations      1
sap abap associate consultant       1
engineer- customer support          1
hr assistant                        1
dcs engineer                        1
Name: Designation, Length: 419, dtype: int64

In [7]:
df.Salary.value_counts()

300000     293
180000     239
200000     205
325000     188
120000     165
          ... 
755000       1
845000       1
1320000      1
1025000      1
725000       1
Name: Salary, Length: 177, dtype: int64

In [8]:
df.Degree.value_counts()

B.Tech/B.E.      3700
MCA               243
M.Tech./M.E.       53
M.Sc. (Tech.)       2
Name: Degree, dtype: int64

In [9]:
df.Specialization.value_counts()

electronics and communication engineering      880
computer science & engineering                 744
information technology                         660
computer engineering                           600
computer application                           244
mechanical engineering                         201
electronics and electrical engineering         196
electronics & telecommunications               121
electrical engineering                          82
electronics & instrumentation eng               32
civil engineering                               29
electronics and instrumentation engineering     27
information science engineering                 27
instrumentation and control engineering         20
electronics engineering                         19
biotechnology                                   15
other                                           13
industrial & production engineering             10
applied electronics and instrumentation          9
chemical engineering           

In [10]:
df['Gender'].value_counts()

m    3041
f     957
Name: Gender, dtype: int64

## info( ) function: This is used to print a concise summary of a DataFrame. This method prints information about a DataFrame including the index dtype and column dtypes, non-null values and memory usage.

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3998 entries, 0 to 3997
Data columns (total 39 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   Unnamed: 0             3998 non-null   object        
 1   ID                     3998 non-null   int64         
 2   Salary                 3998 non-null   int64         
 3   DOJ                    3998 non-null   datetime64[ns]
 4   DOL                    3998 non-null   object        
 5   Designation            3998 non-null   object        
 6   JobCity                3998 non-null   object        
 7   Gender                 3998 non-null   object        
 8   DOB                    3998 non-null   datetime64[ns]
 9   10percentage           3998 non-null   float64       
 10  10board                3998 non-null   object        
 11  12graduation           3998 non-null   int64         
 12  12percentage           3998 non-null   float64       
 13  12b

## FINDING THE MAXIMUM & MINIMUM SALARY

In [12]:
print("Maximum: ", df["Salary"].max())
print("Minimum: ", df["Salary"].min())

Maximum:  4000000
Minimum:  35000


## FINDING THE MAXIMUM SALARY ALONG WITH THE ENTIRE ROW DATA 

In [13]:
df[df['Salary'] == df['Salary'].max()]

Unnamed: 0     ID   Salary        DOJ                  DOL  \
1237      train  48107  4000000 2010-07-01  2015-03-01 00:00:00   
2182      train  41147  4000000 2010-01-01  2011-12-01 00:00:00   

                   Designation    JobCity Gender        DOB  10percentage  \
1237  senior software engineer  New Delhi      m 1987-09-09         90.83   
2182       automation engineer    gurgaon      m 1987-09-13         53.00   

      ... ComputerScience  MechanicalEngg  ElectricalEngg TelecomEngg  \
1237  ...              -1              -1              -1          -1   
2182  ...              -1              -1              -1          -1   

      CivilEngg  conscientiousness agreeableness extraversion  nueroticism  \
1237         -1            -0.7264        0.0328      -1.6502      -0.8778   
2182         -1            -3.3539       -0.4353      -1.0697      -0.0552   

      openess_to_experience  
1237                -0.1295  
2182                -2.6572  

[2 rows x 39 columns]

## Observation:
Males have the highest Salary and Designated with SENIOR SOFTWARE ENGINEER at New Delhi & AUTOMATION ENGINEER at Gurgaon.

## FINDING THE MINIMUN SALARY ALONG WITH THE ENTIRE ROW DATA

In [14]:
df[df['Salary'] == df['Salary'].min()]

Unnamed: 0       ID  Salary        DOJ                  DOL  \
194       train   211840   35000 2012-01-01              present   
2627      train   242100   35000 2012-03-01              present   
3473      train  1272092   35000 2014-05-01  2015-05-01 00:00:00   

           Designation    JobCity Gender        DOB  10percentage  ...  \
194   systems engineer    Gurgaon      f 1989-08-16          86.0  ...   
2627  systems engineer      Noida      m 1989-11-24          88.0  ...   
3473    .net developer  hyderabad      m 1993-01-08          85.3  ...   

     ComputerScience  MechanicalEngg  ElectricalEngg TelecomEngg  CivilEngg  \
194               -1              -1              -1          -1         -1   
2627              -1              -1              -1          -1         -1   
3473             376              -1              -1          -1         -1   

      conscientiousness agreeableness extraversion  nueroticism  \
194             -2.5811        0.0328       0.3817       0.4148   
2627             0.2009        0.0328      -0.9245       0.2973   
3473            -3.7496       -5.6151      -3.8324      -0.4879   

      openess_to_experience  
194                 -1.0774  
2627                -0.1295  
3473                -7.3757  

[3 rows x 39 columns]

## Observation:
1. 2 Males have less Salary Designated as SYSTEMS ENGINEER at Noida & .NET DEVELOPER at Hyderabad.
2. A Female has less Salary Designated as SYSTEMS ENGINEER at Gurgaon.

## loc[ ] function: The loc property is used to access a group of rows and columns by label(s) or a boolean array. It's primarily label based, but may also be used with a boolean array.

## FINDING ENTIRE SALARY DATA GREATER THAN 10 LAKHS

In [15]:
df.loc[df.Salary > 1000000, :]

Unnamed: 0       ID   Salary        DOJ                  DOL  \
3         train   267447  1100000 2011-07-01              present   
92        train  1250429  1500000 2014-11-01  2014-07-01 00:00:00   
123       train   312164  1200000 2010-07-01  2011-07-01 00:00:00   
166       train   323894  1860000 2012-07-01  2014-10-01 00:00:00   
383       train    53921  1110000 2010-07-01  2015-04-01 00:00:00   
394       train    30706  1200000 2011-02-01              present   
500       train   260866  2600000 2011-07-01  2013-04-01 00:00:00   
725       train  1174291  1025000 2014-06-01  2015-07-01 00:00:00   
771       train   134963  1300000 2011-03-01              present   
779       train   325412  3500000 2013-01-01  2014-10-01 00:00:00   
927       train   329920  1300000 2012-11-01  2015-04-01 00:00:00   
943       train   102662  1050000 2010-08-01              present   
961       train   345175  2000000 2012-10-01              present   
1237      train    48107  4000000 2010-07-01  2015-03-01 00:00:00   
1239      train   641560  2020000 2014-06-01  2015-04-01 00:00:00   
1517      train   342930  2050000 2015-04-01              present   
1746      train    46715  1320000 2011-11-01  2015-05-01 00:00:00   
1759      train  1283923  3000000 2014-09-01  2015-03-01 00:00:00   
1982      train   232481  2500000 2012-04-01              present   
2030      train   281074  2300000 2014-03-01              present   
2109      train    40838  1300000 2010-07-01              present   
2152      train   323688  1200000 2012-07-01              present   
2182      train    41147  4000000 2010-01-01  2011-12-01 00:00:00   
2216      train   202950  1800000 2012-12-01  2015-02-01 00:00:00   
2230      train   107796  1200000 2011-01-01              present   
2493      train   644790  1745000 2013-07-01              present   
2541      train  1045685  2000000 2015-06-01              present   
2565      train  1254777  1800000 2015-01-01  2015-04-01 00:00:00   
2764      train   108231  1200000 2010-08-01  2011-06-01 00:00:00   
2880      train   608841  1030000 2013-07-01              present   
3126      train    87319  1210000 2010-10-01  2011-09-01 00:00:00   
3247      train   768298  1500000 2015-06-01  2015-06-01 00:00:00   
3276      train    34551  1100000 2010-06-01              present   
3484      train   615010  2000000 2013-09-01  2014-06-01 00:00:00   
3490      train   803778  2000000 2013-07-01  2014-10-01 00:00:00   
3710      train   271904  1100000 2011-11-01              present   

                     Designation         JobCity Gender        DOB  \
3       senior software engineer         Gurgaon      m 1989-12-05   
92         application developer       Hyderabad      m 1992-01-04   
123             engineer trainee     Maharajganj      m 1988-04-25   
166               field engineer          mumbai      f 1989-10-15   
383             senior developer      Bangalore       m 1988-03-25   
394              design engineer       Bangalore      m 1987-10-02   
500        application developer              -1      m 1989-05-27   
725               branch manager       Bangalore      f 1993-12-02   
771               junior manager           angul      m 1986-05-12   
779    assistant system engineer            Pune      f 1990-08-07   
927        application developer           Dubai      m 1990-07-29   
943                      manager          MUMBAI      m 1989-03-11   
961       software test engineer       bangalore      m 1987-10-13   
1237    senior software engineer       New Delhi      m 1987-09-09   
1239          operations analyst       Bangalore      m 1990-07-20   
1517   software engineer trainee           Noida      m 1991-05-16   
1746       sales account manager           Dubai      m 1987-09-16   
1759                  programmer              -1      m 1992-08-01   
1982      software test engineer              -1      m 1984-09-21   
2030          software developer  Kalmar, Sweden   

In [16]:
df.loc[df.Salary > 1000000, : 'Gender']

Unnamed: 0       ID   Salary        DOJ                  DOL  \
3         train   267447  1100000 2011-07-01              present   
92        train  1250429  1500000 2014-11-01  2014-07-01 00:00:00   
123       train   312164  1200000 2010-07-01  2011-07-01 00:00:00   
166       train   323894  1860000 2012-07-01  2014-10-01 00:00:00   
383       train    53921  1110000 2010-07-01  2015-04-01 00:00:00   
394       train    30706  1200000 2011-02-01              present   
500       train   260866  2600000 2011-07-01  2013-04-01 00:00:00   
725       train  1174291  1025000 2014-06-01  2015-07-01 00:00:00   
771       train   134963  1300000 2011-03-01              present   
779       train   325412  3500000 2013-01-01  2014-10-01 00:00:00   
927       train   329920  1300000 2012-11-01  2015-04-01 00:00:00   
943       train   102662  1050000 2010-08-01              present   
961       train   345175  2000000 2012-10-01              present   
1237      train    48107  4000000 2010-07-01  2015-03-01 00:00:00   
1239      train   641560  2020000 2014-06-01  2015-04-01 00:00:00   
1517      train   342930  2050000 2015-04-01              present   
1746      train    46715  1320000 2011-11-01  2015-05-01 00:00:00   
1759      train  1283923  3000000 2014-09-01  2015-03-01 00:00:00   
1982      train   232481  2500000 2012-04-01              present   
2030      train   281074  2300000 2014-03-01              present   
2109      train    40838  1300000 2010-07-01              present   
2152      train   323688  1200000 2012-07-01              present   
2182      train    41147  4000000 2010-01-01  2011-12-01 00:00:00   
2216      train   202950  1800000 2012-12-01  2015-02-01 00:00:00   
2230      train   107796  1200000 2011-01-01              present   
2493      train   644790  1745000 2013-07-01              present   
2541      train  1045685  2000000 2015-06-01              present   
2565      train  1254777  1800000 2015-01-01  2015-04-01 00:00:00   
2764      train   108231  1200000 2010-08-01  2011-06-01 00:00:00   
2880      train   608841  1030000 2013-07-01              present   
3126      train    87319  1210000 2010-10-01  2011-09-01 00:00:00   
3247      train   768298  1500000 2015-06-01  2015-06-01 00:00:00   
3276      train    34551  1100000 2010-06-01              present   
3484      train   615010  2000000 2013-09-01  2014-06-01 00:00:00   
3490      train   803778  2000000 2013-07-01  2014-10-01 00:00:00   
3710      train   271904  1100000 2011-11-01              present   

                     Designation         JobCity Gender  
3       senior software engineer         Gurgaon      m  
92         application developer       Hyderabad      m  
123             engineer trainee     Maharajganj      m  
166               field engineer          mumbai      f  
383             senior developer      Bangalore       m  
394              design engineer       Bangalore      m  
500        application developer              -1      m  
725               branch manager       Bangalore      f  
771               junior manager           angul      m  
779    assistant system engineer            Pune      f  
927        application developer           Dubai      m  
943                      manager          MUMBAI      m  
961       software test engineer       bangalore      m  
1237    senior software engineer       New Delhi      m  
1239          operations analyst       Bangalore      m  
1517   software engineer trainee           Noida      m  
1746       sales account manager           Dubai      m  
1759                  programmer              -1      m  
1982      software test engineer              -1      m  
2030          software developer  Kalmar, Sweden      m  
2109             senior engineer         Gurgaon      m  
2152  embedded software engineer       Bangalore      m  
2182         automation engineer         gurgaon      m  
2216   client services associate       Bangalore      f  
2

## Observation:
29 Males & 7 females have Salary > 10 Lakhs where most of them are SOFTWARE ENGINEERS.

## FINDING SALARY DATA GREATER THAN 10 LAKHS & COLLEGE GPA GREATER THAN 80 WITH SPECIALIZATION 

In [17]:
df.loc[(df.Salary > 1000000) & (df.collegeGPA > 80), ['Specialization']]

Specialization
166                        electrical engineering
771       instrumentation and control engineering
943                        information technology
2152       electronics and electrical engineering
2493  electronics and instrumentation engineering
2541               computer science & engineering
2880                         computer engineering
3276    electronics and communication engineering

## Observation:
8 Persons have Salary > 10 Lakhs & College-GPA > 80

## FINDING THE SALARIES EXCATLY EQUAL TO 10LAKHS, 11LAKHS, 12LAKHS, 18LAKHS, 20LAKHS, 30LAKHS WITH GENDER

In [18]:
df.loc[df.Salary == 1000000, : 'Gender']

Unnamed: 0       ID   Salary        DOJ      DOL        Designation  \
395       train   279824  1000000 2012-08-01  present      asst. manager   
2412      train    42943  1000000 2010-09-01  present  assistant manager   
2472      train  1283372  1000000 2015-01-01  present  software engineer   

          JobCity Gender  
395   Muzaffarpur      m  
2412     Vadodara      m  
2472        Noida      m

In [19]:
df.loc[df.Salary == 1100000, : 'Gender']

Unnamed: 0      ID   Salary        DOJ      DOL  \
3         train  267447  1100000 2011-07-01  present   
3276      train   34551  1100000 2010-06-01  present   
3710      train  271904  1100000 2011-11-01  present   

                   Designation        JobCity Gender  
3     senior software engineer        Gurgaon      m  
3276           design engineer  Greater Noida      m  
3710  senior software engineer      Bangalore      m

In [20]:
df.loc[df.Salary == 1200000, : 'Gender']

Unnamed: 0      ID   Salary        DOJ                  DOL  \
123       train  312164  1200000 2010-07-01  2011-07-01 00:00:00   
394       train   30706  1200000 2011-02-01              present   
2152      train  323688  1200000 2012-07-01              present   
2230      train  107796  1200000 2011-01-01              present   
2764      train  108231  1200000 2010-08-01  2011-06-01 00:00:00   

                     Designation      JobCity Gender  
123             engineer trainee  Maharajganj      m  
394              design engineer    Bangalore      m  
2152  embedded software engineer    Bangalore      m  
2230                    engineer       Mumbai      m  
2764          office coordinator    Hyderabad      f

In [21]:
df.loc[df.Salary == 1800000, : 'Gender']

Unnamed: 0       ID   Salary        DOJ                  DOL  \
2216      train   202950  1800000 2012-12-01  2015-02-01 00:00:00   
2565      train  1254777  1800000 2015-01-01  2015-04-01 00:00:00   

                    Designation    JobCity Gender  
2216  client services associate  Bangalore      f  
2565       salesforce developer  Panchkula      f

In [22]:
df.loc[df.Salary == 2000000, : 'Gender']

Unnamed: 0       ID   Salary        DOJ                  DOL  \
961       train   345175  2000000 2012-10-01              present   
2541      train  1045685  2000000 2015-06-01              present   
3484      train   615010  2000000 2013-09-01  2014-06-01 00:00:00   
3490      train   803778  2000000 2013-07-01  2014-10-01 00:00:00   

                 Designation    JobCity Gender  
961   software test engineer  bangalore      m  
2541          data scientist     LONDON      m  
3484           it technician      Noida      m  
3490          technical lead       Pune      m

In [23]:
df.loc[df.Salary == 3000000, : 'Gender']

Unnamed: 0       ID   Salary        DOJ                  DOL Designation  \
1759      train  1283923  3000000 2014-09-01  2015-03-01 00:00:00  programmer   

     JobCity Gender  
1759      -1      m

## Observations:
1. 3 Males have Salary excatly equal to 10 Lakhs designated with ASSISTANT MANAGER & SOFTWARE ENGINEER.
2. 3 Males have Salary excatly equal to 11 Lakhs designated with SENIOR SOFTWARE ENGINEER & DESIGN ENGINEER.
3. 4 Males and a Female have Salary excatly equal to 12 Lakhs.
4. 2 Females have Salary excatly equal to 18 Lakhs designated with CLIENT SERVICES ASSOCIATE & SALES FORCE DEVELOPER.
5. 4 Males have Salary excatly equal to 20 Lakhs where one of them is designated as a DATA SCIENTIST working at LONDON.  
6. 1 Male have Salary excatly equal to 30 Lakhs designated as PROGRAMMER.

## FINDING THE COLLEGE TIER

In [24]:
df[df['CollegeTier'] == df['CollegeTier'].max()]

Unnamed: 0       ID  Salary        DOJ                  DOL  \
0         train   203097  420000 2012-06-01              present   
1         train   579905  500000 2013-09-01              present   
2         train   810601  325000 2014-06-01              present   
4         train   343523  200000 2014-03-01  2015-03-01 00:00:00   
5         train  1027655  300000 2014-06-01              present   
...         ...      ...     ...        ...                  ...   
3993      train    47916  280000 2011-10-01  2012-10-01 00:00:00   
3994      train   752781  100000 2013-07-01  2013-07-01 00:00:00   
3995      train   355888  320000 2013-07-01              present   
3996      train   947111  200000 2014-07-01  2015-01-01 00:00:00   
3997      train   324966  400000 2013-02-01              present   

                      Designation           JobCity Gender        DOB  \
0         senior quality engineer         Bangalore      f 1990-02-19   
1               assistant manager            Indore      m 1989-10-04   
2                systems engineer           Chennai      f 1992-08-03   
4                             get           Manesar      m 1991-02-27   
5                 system engineer         Hyderabad      m 1992-07-02   
...                           ...               ...    ...        ...   
3993            software engineer        New Delhi       m 1987-04-15   
3994             technical writer         Hyderabad      f 1992-08-27   
3995  associate software engineer         Bangalore      m 1991-07-03   
3996           software developer  Asifabadbanglore      f 1992-03-20   
3997      senior systems engineer           Chennai      f 1991-02-26   

      10percentage  ... ComputerScience  MechanicalEngg  ElectricalEngg  \
0            84.30  ...              -1              -1              -1   
1            85.40  ...              -1              -1              -1   
2            85.00  ...              -1              -1              -1   
4            78.00  ...              -1              -1              -1   
5            89.92  ...             407              -1              -1   
...            ...  ...             ...             ...             ...   
3993         52.09  ...              -1              -1              -1   
3994         90.00  ...              -1              -1              -1   
3995         81.86  ...              -1              -1              -1   
3996         78.72  ...             438              -1              -1   
3997         70.60  ...              -1              -1              -1   

     TelecomEngg  CivilEngg  conscientiousness agreeableness extraversion  \
0             -1         -1             0.9737        0.8128       0.5269   
1             -1         -1            -0.7335        0.3789       1.2396   
2             -1         -1             0.2718        1.7109       0.1637   
4             -1         -1            -0.8810       -0.2793      -1.0697   
5             -1         -1            -0.3027       -0.6201      -2.2954   
...          ...        ...                ...           ...          ...   
3993          -1         -1            -0.1082        0.3448       0.2366   
3994          -1         -1            -0.3027        0.8784       0.9322   
3995          -1         -1            -1.5765       -1.5273      -1.5051   
3996          -1         -1            -0.1590        0.0459      -0.4511   
3997          -1         -1            -1.1128       -0.2793      -0.6343   

      nueroticism  openess_to_experience  
0         1.35490                -0.4455  
1        -0.10760                 0.8637  
2        -0.86820                 0.6721  
4         0.09163                -0.1295  
5        -0.74150                -0.8608  
...           ...                    ...  
3993      0.64980                -0.9194  
3994      0.77980                -0.0943  
3995     -1.31840                -0.7615  
3996     -0.36120                -0.0943  
3997      1.

In [25]:
df[df['CollegeTier'] == df['CollegeTier'].min()]

Unnamed: 0       ID   Salary        DOJ                  DOL  \
3         train   267447  1100000 2011-07-01              present   
7         train   912934   400000 2014-07-01  2015-07-01 00:00:00   
8         train   552574   600000 2013-07-01              present   
11        train   190393   450000 2011-08-01  2014-10-01 00:00:00   
32        train   580616   360000 2013-10-01              present   
...         ...      ...      ...        ...                  ...   
3912      train   231229   730000 2013-07-01              present   
3913      train   913446   370000 2014-06-01  2015-04-01 00:00:00   
3914      train   816649   220000 2014-10-01  2015-03-01 00:00:00   
3972      train  1044603   180000 2015-03-01  2015-05-01 00:00:00   
3979      train   212055   550000 2013-07-01  2014-04-01 00:00:00   

                               Designation     JobCity Gender        DOB  \
3                 senior software engineer     Gurgaon      m 1989-12-05   
7                      mechanical engineer   Bangalore      m 1992-05-27   
8                      electrical engineer       Noida      m 1991-09-17   
11                 senior systems engineer        Pune      m 1990-02-10   
32                   application developer       Noida      f 1991-09-14   
...                                    ...         ...    ...        ...   
3912                    research scientist       Pune       m 1989-11-15   
3913         software development engineer  Coimbatore      m 1992-12-24   
3914  field business development associate   New Delhi      m 1992-04-04   
3972                     help desk analyst       Noida      m 1992-11-30   
3979                     software engineer   Bangalore      m 1989-07-22   

      10percentage  ... ComputerScience  MechanicalEngg  ElectricalEngg  \
3            85.60  ...              -1              -1              -1   
7            92.00  ...              -1             469              -1   
8            90.00  ...              -1              -1              -1   
11           81.00  ...              -1              -1              -1   
32           87.00  ...              -1              -1              -1   
...            ...  ...             ...             ...             ...   
3912         84.67  ...              -1              -1              -1   
3913         77.80  ...             438              -1              -1   
3914         71.80  ...              -1              -1              -1   
3972         68.20  ...             407              -1              -1   
3979         69.16  ...              -1              -1              -1   

     TelecomEngg  CivilEngg  conscientiousness agreeableness extraversion  \
3             -1         -1             0.0464        0.3448      -0.3440   
7             -1         -1            -0.0154        1.2114       0.0100   
8             -1         -1            -0.1590        0.5454      -0.6048   
11            -1         -1            -2.2720        1.1248       0.8171   
32            -1         -1            -0.7335       -0.4536      -1.0659   
...          ...        ...                ...           ...          ...   
3912          -1         -1            -1.3447       -1.0593       0.6720   
3913          -1         -1             0.2718        0.0459       0.1637   
3914         260         -1            -0.7335       -0.6201       0.6248   
3972          -1         -1            -1.1644        0.2124      -0.2974   
3979          -1         -1            -0.5719        0.5008      -0.4891   

      nueroticism  openess_to_experience  
3         -0.4078                -0.9194  
7          0.1459                 1.2470  
8         -0.7415                -0.2859  
11        -0.4078                -0.4455  
32         1.1601                -0.4776  
...           ...                    ...  
3912       1.0024                -1.7093  
3913       1.4136                -2.2021  
3914       1.5404                -0.6692  
3972   

## Observations:
1. 297 Persons are from TIER - 1 COLLEGE.
2. 3701 Persons are from TIER - 2 COLLEGE.

## FINDING THE 10TH, 12TH, COLLEGE GPA WITH SPECIALIZATION & GENDER 

In [26]:
A = df[['10percentage', '12percentage', 'collegeGPA','Specialization', 'Gender']]
print(type(A))
A

<class 'pandas.core.frame.DataFrame'>


10percentage  12percentage  collegeGPA  \
0            84.30         95.80       78.00   
1            85.40         85.00       70.06   
2            85.00         68.20       70.00   
3            85.60         83.60       74.64   
4            78.00         76.80       73.90   
...            ...           ...         ...   
3993         52.09         55.50       61.50   
3994         90.00         93.00       77.30   
3995         81.86         65.50       70.00   
3996         78.72         69.88       70.42   
3997         70.60         68.00       68.00   

                                 Specialization Gender  
0                          computer engineering      f  
1     electronics and communication engineering      m  
2                        information technology      f  
3                          computer engineering      m  
4     electronics and communication engineering      m  
...                                         ...    ...  
3993                     information technology      m  
3994  electronics and communication engineering      f  
3995                       computer engineering      m  
3996             computer science & engineering      f  
3997                     information technology      f  

[3998 rows x 5 columns]

In [28]:
DF = pd.read_excel(r'C:\Users\admin\Desktop\KANAV BANSAL\KANAV TASKS\TASK - II - EDA - AMCAT DATA/reports.xlsx')
DF = DF[['English', 'Logical', 'Quant', 'Specialization', 'Gender']]
print(type(DF))
DF

<class 'pandas.core.frame.DataFrame'>


English  Logical  Quant                             Specialization  \
0         515      585    525                       computer engineering   
1         695      610    780  electronics and communication engineering   
2         615      545    370                     information technology   
3         635      585    625                       computer engineering   
4         545      625    465  electronics and communication engineering   
...       ...      ...    ...                                        ...   
3993      365      334    475                     information technology   
3994      415      410    535  electronics and communication engineering   
3995      475      475    465                       computer engineering   
3996      450      410    320             computer science & engineering   
3997      565      515    464                     information technology   

     Gender  
0         f  
1         m  
2         f  
3         m  
4         m  
...     ...  
3993      m  
3994      f  
3995      m  
3996      f  
3997      f  

[3998 rows x 5 columns]

## groupby( ) function: It is used to split the data into groups based on some criteria. Pandas objects can be split on any of their axes. The abstract definition of grouping is to provide a mapping of labels to group names.

In [29]:
new_df = df.groupby('Specialization')
new_df.first()

Unnamed: 0       ID  Salary  \
Specialization                                                            
aeronautical engineering                         train   593198  120000   
applied electronics and instrumentation          train  1044453  310000   
automobile/automotive engineering                train   676834  380000   
biomedical engineering                           train   280764  145000   
biotechnology                                    train   318523  400000   
ceramic engineering                              train   945679  335000   
chemical engineering                             train   914981  200000   
civil engineering                                train  1089283  110000   
computer and communication engineering           train  1111630  120000   
computer application                             train   874596  250000   
computer engineering                             train   203097  420000   
computer networking                              train   353442  565000   
computer science                                 train  1250504  400000   
computer science & engineering                   train   947847  300000   
computer science and technology                  train   983664  300000   
control and instrumentation engineering          train  1081403  305000   
electrical and power engineering                 train   978959  240000   
electrical engineering                           train  1082717  350000   
electronics                                      train   578157   40000   
electronics & instrumentation eng                train  1059903  300000   
electronics & telecommunications                 train   606655  270000   
electronics and communication engineering        train   579905  500000   
electronics and computer engineering             train   852310  120000   
electronics and electrical engineering           train   552574  600000   
electronics and instrumentation engineering      train   543139  110000   
electronics engineering                          train   431838  330000   
embedded systems technology                      train   918414  200000   
industrial & management engineering              train  1229820  320000   
industrial & production engineering              train   134828  180000   
industrial engineering                           train   711754  390000   
information & communication technology           train   896880  450000   
information science                              train   552431  460000   
information science engineering                  train   991197  100000   
information technology                           train   810601  325000   
instrumentation and control engineering          train  1279958  300000   
instrumentation engineering                      train   708034  260000   
internal combustion engine                       train  1067508  360000   
mechanical & production engineering              train   242799  100000   
mechanical and automation                        train   320969  500000   
mechanical engineering                           train   912934  400000   
mechatronics                                     train   879126  350000   
metallurgical engineering                        train  1045625  300000   
other                                            train   611740  300000   
polymer technology                               train   962376  700000   
power systems and automation                     train   652877  100000   
telecommunication engineering                    train   897769  400000   

                                                   DOJ                  DOL  \
Specialization                                                                
aeronautical engineering                    2013-06-01  2014-04-01 00:00:00   
applied electronics and instrumentation     2014-02-01  2014-05-01 00:00:00   
automobile/automotive engineering           2014-01-01              present   
biomedical engineering                      2011

In [30]:
new_df.groups.keys()

dict_keys(['aeronautical engineering', 'applied electronics and instrumentation', 'automobile/automotive engineering', 'biomedical engineering', 'biotechnology', 'ceramic engineering', 'chemical engineering', 'civil engineering', 'computer and communication engineering', 'computer application', 'computer engineering', 'computer networking', 'computer science', 'computer science & engineering', 'computer science and technology', 'control and instrumentation engineering', 'electrical and power engineering', 'electrical engineering', 'electronics', 'electronics & instrumentation eng', 'electronics & telecommunications', 'electronics and communication engineering', 'electronics and computer engineering', 'electronics and electrical engineering', 'electronics and instrumentation engineering', 'electronics engineering', 'embedded systems technology', 'industrial & management engineering', 'industrial & production engineering', 'industrial engineering', 'information & communication technology

In [31]:
new_df.get_group('computer engineering')

Unnamed: 0      ID   Salary        DOJ                  DOL  \
0         train  203097   420000 2012-06-01              present   
3         train  267447  1100000 2011-07-01              present   
10        train   87291   600000 2011-04-01  2015-04-01 00:00:00   
19        train  466888   325000 2014-09-01              present   
20        train  140069   320000 2010-11-01  2012-09-01 00:00:00   
...         ...     ...      ...        ...                  ...   
3968      train  469012   360000 2013-05-01  2015-01-01 00:00:00   
3979      train  212055   550000 2013-07-01  2014-04-01 00:00:00   
3984      train  305041   480000 2011-12-01              present   
3985      train  358393   430000 2013-03-01              present   
3995      train  355888   320000 2013-07-01              present   

                      Designation    JobCity Gender        DOB  10percentage  \
0         senior quality engineer  Bangalore      f 1990-02-19         84.30   
3        senior software engineer    Gurgaon      m 1989-12-05         85.60   
10           senior php developer  Bangalore      m 1989-06-24         88.60   
19              software engineer       Pune      f 1990-11-30         79.00   
20              software engineer  Bangalore      f 1988-07-25         91.20   
...                           ...        ...    ...        ...           ...   
3968            software engineer  Bangalore      f 1990-09-30         81.00   
3979            software engineer  Bangalore      m 1989-07-22         69.16   
3984            software engineer    Gurgaon      f 1990-01-18         89.80   
3985              system engineer    Gurgaon      f 1990-05-14         90.40   
3995  associate software engineer  Bangalore      m 1991-07-03         81.86   

      ... ComputerScience  MechanicalEngg  ElectricalEngg TelecomEngg  \
0     ...              -1              -1              -1          -1   
3     ...              -1              -1              -1          -1   
10    ...              -1              -1              -1          -1   
19    ...              -1              -1              -1          -1   
20    ...              -1              -1              -1          -1   
...   ...             ...             ...             ...         ...   
3968  ...              -1              -1              -1          -1   
3979  ...              -1              -1              -1          -1   
3984  ...              -1              -1              -1          -1   
3985  ...              -1              -1              -1          -1   
3995  ...              -1              -1              -1          -1   

      CivilEngg  conscientiousness agreeableness  extraversion  nueroticism  \
0            -1             0.9737        0.8128        0.5269       1.3549   
3            -1             0.0464        0.3448       -0.3440      -0.4078   
10           -1             0.0464        0.8128        0.0914       0.1798   
19           -1             0.8192        0.2668       -0.2714      -0.4078   
20           -1            -0.1082        0.9688        0.5269      -0.2902   
...         ...                ...           ...           ...          ...   
3968         -1             0.0464        0.3448       -0.3440       0.6204   
3979         -1            -0.5719        0.5008       -0.4891       0.4148   
3984         -1            -0.2628        0.1888        0.3817      -0.2902   
3985         -1             0.2782        0.3448        0.0914      -1.1422   
3995         -1            -1.5765       -1.5273       -1.5051      -1.3184   

      openess_to_experience  
0                   -0.4455  
3                   -0.9194  
10                   0.0284  
19                  -0.1295  
20                   0.3444  
...                     ...  
3968                 0.3444  
3979                -1.2354  
3984                 1.6082  
3985                -0.6035  
3995                -0.7615  

[600 rows x 38 columns]

## Observation:
There are 600 Persons from COMPUTER SCIENCE ENGINEERING stream, where most of them are SOFTWARE ENGINEERS.

In [32]:
new_df.get_group('electronics and communication engineering')

Unnamed: 0       ID  Salary        DOJ                  DOL  \
1         train   579905  500000 2013-09-01              present   
4         train   343523  200000 2014-03-01  2015-03-01 00:00:00   
5         train  1027655  300000 2014-06-01              present   
9         train  1203363  230000 2014-07-01              present   
13        train   431800  200000 2014-01-01              present   
...         ...      ...     ...        ...                  ...   
3972      train  1044603  180000 2015-03-01  2015-05-01 00:00:00   
3983      train  1076441  315000 2015-01-01              present   
3987      train   439787  280000 2012-11-01              present   
3990      train   204287  480000 2012-02-01              present   
3994      train   752781  100000 2013-07-01  2013-07-01 00:00:00   

                  Designation    JobCity Gender        DOB  10percentage  ...  \
1           assistant manager     Indore      m 1989-10-04         85.40  ...   
4                         get    Manesar      m 1991-02-27         78.00  ...   
5             system engineer  Hyderabad      m 1992-07-02         89.92  ...   
9            project engineer    Kolkata      m 1993-06-13         77.00  ...   
13                 qa analyst    Gurgaon      m 1990-10-23         60.80  ...   
...                       ...        ...    ...        ...           ...  ...   
3972        help desk analyst      Noida      m 1992-11-30         68.20  ...   
3983     system administrator  Hyderabad      m 1991-06-20         88.10  ...   
3987         network engineer  New Delhi      f 1990-01-16         86.70  ...   
3990  senior systems engineer  Hyderabad      f 1989-09-04         88.00  ...   
3994         technical writer  Hyderabad      f 1992-08-27         90.00  ...   

     ComputerScience  MechanicalEngg  ElectricalEngg TelecomEngg  CivilEngg  \
1                 -1              -1              -1          -1         -1   
4                 -1              -1              -1          -1         -1   
5                407              -1              -1          -1         -1   
9                 -1              -1              -1          -1         -1   
13                -1              -1              -1          -1         -1   
...              ...             ...             ...         ...        ...   
3972             407              -1              -1          -1         -1   
3983              -1              -1              -1          -1         -1   
3987              -1              -1              -1          -1         -1   
3990              -1              -1              -1         233         -1   
3994              -1              -1              -1          -1         -1   

      conscientiousness agreeableness  extraversion  nueroticism  \
1               -0.7335        0.3789        1.2396     -0.10760   
4               -0.8810       -0.2793       -1.0697      0.09163   
5               -0.3027       -0.6201       -2.2954     -0.74150   
9               -1.3080        0.5454       -0.9122      0.90660   
13               0.3555        0.0328       -0.5980     -0.29020   
...                 ...           ...           ...          ...   
3972            -1.1644        0.2124       -0.2974      0.01920   
3983            -0.1590        0.8784        0.4711      0.27270   
3987            -1.4992       -1.8393       -0.7794      1.47240   
3990             0.6646        0.3448        0.3817     -1.34780   
3994            -0.3027        0.8784        0.9322      0.77980   

      openess_to_experience  
1                    0.8637  
4                   -0.1295  
5                   -0.8608  
9                    0.0973  
13                  -1.2354  
...                     ...  
3972                -1.0524  
3983                 0.6721  
3987                -2.3017  
3990                 0.8183  
3994                -0.0943  

[880 rows x 38 columns]

## Observation:
There are 880 Persons from ELECTRONICS AND COMMUNICATION ENGINEERING stream, where most of them are SYSTEM ENGINEERS.

In [33]:
new_df.get_group('electrical engineering')

Unnamed: 0       ID  Salary        DOJ                  DOL  \
53        train  1082717  350000 2014-09-01              present   
70        train   801755  320000 2014-03-01              present   
82        train    87067  215000 2014-10-01  2015-03-01 00:00:00   
125       train   796063  300000 2014-11-01  2015-03-01 00:00:00   
144       train   612253  385000 2013-06-01              present   
...         ...      ...     ...        ...                  ...   
3733      train   962526  700000 2014-06-01              present   
3747      train   265160  195000 2011-06-01  2012-01-01 00:00:00   
3832      train   625434  565000 2013-10-01  2015-04-01 00:00:00   
3839      train   644828  360000 2013-09-01              present   
3897      train   329618  370000 2012-11-01              present   

                         Designation      JobCity Gender        DOB  \
53         assistant system engineer       Mumbai      m 1991-08-06   
70                   system engineer      Chennai      m 1992-12-12   
82    entry level management trainee    New Delhi      f 1987-03-15   
125               programmer analyst      Kolkata      m 1992-03-19   
144                         engineer           -1      m 1992-06-12   
...                              ...          ...    ...        ...   
3733               software engineer    Hyderabad      m 1991-11-05   
3747             electrical engineer      Kolkata      m 1989-09-01   
3832                        engineer       Haldia      m 1991-10-28   
3839                             ase  bhubaneswar      f 1992-03-11   
3897               software engineer      Kolkata      m 1990-10-20   

      10percentage  ... ComputerScience  MechanicalEngg  ElectricalEngg  \
53           89.25  ...              -1              -1             484   
70           89.00  ...              -1              -1             606   
82           69.00  ...              -1              -1              -1   
125          90.00  ...              -1              -1             500   
144          64.56  ...              -1              -1              -1   
...            ...  ...             ...             ...             ...   
3733         81.50  ...              -1              -1             446   
3747         71.73  ...              -1              -1              -1   
3832         92.83  ...              -1              -1             660   
3839         90.80  ...              -1              -1             606   
3897         83.25  ...              -1              -1              -1   

     TelecomEngg  CivilEngg  conscientiousness agreeableness  extraversion  \
53            -1         -1            -0.1590       -0.1206       -0.6048   
70            -1         -1            -1.4517       -1.4526       -1.2196   
82            -1         -1            -0.4173        0.1888       -0.4891   
125           -1         -1             0.1282       -0.2871        1.3933   
144           -1         -1             0.1282        0.0459        0.1637   
...          ...        ...                ...           ...           ...   
3733          -1         -1            -1.5953       -0.1206       -0.2974   
3747          -1         -1             0.5100        1.2808        0.6720   
3832          -1         -1             0.1282        0.0459       -0.1437   
3839          -1         -1             1.5644        0.5454       -0.2974   
3897          -1         -1            -1.1128       -1.0593       -0.3440   

      nueroticism  openess_to_experience  
53       -1.62890                -1.4356  
70        1.79400                -2.3937  
82        0.41480                 0.8183  
125       0.65300                -0.6692  
144       0.27270                -0.2859  
...           ...                    ...  
3733     -0.99500                -0.8608  
3747      0.88480                 0.3444  
3832     -0.23440                 0.0973  
3839     -1.37530                 1.0554  
3897     -0.17277       

## Observation:
There are 82 Persons from ELECTRICAL AND ELECTRONICS ENGINEERING stream, where most of them are SOFTWARE ENGINEERS.

In [34]:
new_df.get_group('mechanical engineering')

Unnamed: 0       ID   Salary        DOJ                  DOL  \
7         train   912934   400000 2014-07-01  2015-07-01 00:00:00   
40        train   245887   120000 2011-06-01  2012-03-01 00:00:00   
45        train  1230582   110000 2014-09-01  2015-01-01 00:00:00   
63        train   636871   150000 2014-10-01  2015-04-01 00:00:00   
123       train   312164  1200000 2010-07-01  2011-07-01 00:00:00   
...         ...      ...      ...        ...                  ...   
3849      train  1002611   325000 2014-11-01              present   
3909      train   628271   210000 2013-08-01              present   
3916      train   530146   360000 2013-06-01  2014-12-01 00:00:00   
3951      train   885459   145000 2014-05-01              present   
3980      train   197796   150000 2011-07-01  2012-07-01 00:00:00   

              Designation      JobCity Gender        DOB  10percentage  ...  \
7     mechanical engineer    Bangalore      m 1992-05-27         92.00  ...   
40           site manager       Jhansi      m 1989-05-04         66.60  ...   
45    production engineer      Chennai      m 1992-10-12         95.80  ...   
63     management trainee      Bhiwadi      m 1992-01-10         60.66  ...   
123      engineer trainee  Maharajganj      m 1988-04-25         59.80  ...   
...                   ...          ...    ...        ...           ...  ...   
3849     systems engineer    Hyderabad      m 1992-09-09         89.17  ...   
3909  production engineer  Navi Mumbai      m 1992-10-25         67.00  ...   
3916   management trainee       Hissar      m 1991-07-14         79.80  ...   
3951  mechanical engineer        Karad      m 1992-08-27         93.07  ...   
3980                  get      haryana      m 1986-08-05         84.00  ...   

     ComputerScience  MechanicalEngg  ElectricalEngg TelecomEngg  CivilEngg  \
7                 -1             469              -1          -1         -1   
40                -1             286              -1          -1         -1   
45                -1             253              -1          -1         -1   
63                -1             446              -1          -1         -1   
123               -1             206              -1          -1         -1   
...              ...             ...             ...         ...        ...   
3849              -1             623              -1          -1         -1   
3909              -1             420              -1          -1         -1   
3916              -1             393              -1          -1         -1   
3951              -1             407              -1          -1         -1   
3980              -1             473              -1          -1         -1   

      conscientiousness agreeableness  extraversion  nueroticism  \
7               -0.0154        1.2114        0.0100       0.1459   
40              -0.4173        0.3448        0.0914      -0.4078   
45              -0.5899       -1.1196        0.4711      -0.7415   
63              -2.2351       -2.6847       -2.6028       0.3995   
123              0.2009        1.1248        1.1074      -1.1128   
...                 ...           ...           ...          ...   
3849             0.1282        0.5454       -0.4511      -1.6289   
3909             0.7027       -0.0873       -0.0319       0.6530   
3916             0.3941        0.1888       -0.6343       0.0917   
3951            -0.3027       -1.4526       -0.2974       1.2869   
3980            -1.4992        0.5008       -0.9245      -0.6428   

      openess_to_experience  
7                    1.2470  
40                   0.6603  
45                  -1.4356  
63                  -5.4770  
123                  0.9763  
...                     ...  
3849                 0.6721  
3909                 0.1670  
3916                -1.3934  
3951                -1.4356  
3980                -0.6035  

[201 rows x 38 columns]

## Observation:
There are 201 Persons from MECHANICAL ENGINEERING stream, where most of them are PRODUCTION ENGINEERS.

In [35]:
new_df.get_group('civil engineering')

Unnamed: 0       ID  Salary        DOJ                  DOL  \
172       train  1089283  110000 2014-06-01  2015-02-01 00:00:00   
331       train   341114  700000 2014-04-01              present   
403       train   795799  405000 2014-07-01              present   
453       train  1253724  300000 2015-06-01              present   
492       train   795862  525000 2014-09-01              present   
493       train   609066  720000 2014-02-01              present   
605       train   319130  350000 2012-06-01  2015-04-01 00:00:00   
607       train   642404  600000 2013-07-01              present   
896       train   285145  300000 2012-06-01  2014-09-01 00:00:00   
1108      train  1284153  225000 2014-04-01  2014-08-01 00:00:00   
1583      train   637094  180000 2014-10-01              present   
1678      train   877224  300000 2014-12-01              present   
1858      train   693253  330000 2013-12-01  2014-04-01 00:00:00   
1879      train  1089302  375000 2014-08-01              present   
2242      train   796983  415000 2014-06-01              present   
2419      train   877241  345000 2014-07-01  2015-01-01 00:00:00   
2469      train   333439  290000 2012-08-01              present   
2571      train   927284  330000 2014-08-01              present   
2591      train   652269  180000 2013-12-01  2014-05-01 00:00:00   
2690      train   946109  180000 2014-03-01              present   
3015      train    38335  720000 2010-07-01              present   
3332      train   950502  150000 2014-06-01  2015-03-01 00:00:00   
3513      train   988953  310000 2014-06-01              present   
3821      train   810998  300000 2014-07-01              present   
3865      train  1095963  300000 2014-08-01              present   
3868      train   618243  320000 2013-06-01              present   
3905      train   708980  295000 2015-03-01              present   
3961      train   230702  700000 2011-07-01  2014-09-01 00:00:00   
3992      train   344407  800000 2014-04-01  2015-04-01 00:00:00   

                 Designation     JobCity Gender        DOB  10percentage  ...  \
172      assistant professor         Una      f 1993-04-15         70.00  ...   
331        assistant manager        pune      f 1990-02-16         82.50  ...   
403        planning engineer       Noida      m 1991-10-06         93.40  ...   
453        software engineer        pune      m 1992-10-29         90.15  ...   
492         business analyst       Noida      m 1992-09-11         92.00  ...   
493             data analyst     Gurgaon      m 1991-08-24         87.80  ...   
605   software test engineer   Bangalore      m 1990-12-07         87.11  ...   
607                 engineer   New Delhi      m 1992-07-01         92.00  ...   
896         project engineer   Hyderabad      f 1989-08-02         90.83  ...   
1108      assistant engineer    Dehradun      m 1993-09-16         73.00  ...   
1583      technical engineer   New Delhi      m 1993-07-10         67.30  ...   
1678      management trainee        Pune      f 1992-05-10         93.38  ...   
1858        project engineer          -1      m 1992-08-05         62.00  ...   
1879  junior research fellow  Chandigarh      m 1992-03-18         84.60  ...   
2242       assistant manager       PATNA      m 1993-01-20         74.80  ...   
2419         design engineer        Pune      m 1992-05-10         95.54  ...   
2469          civil engineer   Joshimath      m 1991-09-20         67.00  ...   
2571        business analyst   Bangalore      f 1992-09-27         90.00  ...   
2591                lecturer     Chennai      m 1991-12-27         76.00  ...   
2690          civil engineer   Ahmedabad      m 1992-10-10         79.20  ...   
3015          civil engineer       patna      m 1985-11-17         82.00  ...   
3332         design engineer     Chennai      f 1993-02-14         87.80  ...   
3513      assistant engineer      Mumbai      m 1992-04-02         77.20  ...   
3821                

## Observation:
There are 29 Persons from CIVIL ENGINEERING stream, where most of them are DESIGN ENGINEERS.

In [36]:
new_df.get_group('information technology')

Unnamed: 0       ID  Salary        DOJ                  DOL  \
2         train   810601  325000 2014-06-01              present   
11        train   190393  450000 2011-08-01  2014-10-01 00:00:00   
15        train   536517  350000 2013-08-01              present   
21        train   339689  200000 2012-08-01  2013-12-01 00:00:00   
29        train   472956  220000 2014-11-01  2014-04-01 00:00:00   
...         ...      ...     ...        ...                  ...   
3976      train  1006997  300000 2015-01-01              present   
3977      train   217036  415000 2011-09-01              present   
3978      train  1274658  320000 2013-08-01  2014-01-01 00:00:00   
3993      train    47916  280000 2011-10-01  2012-10-01 00:00:00   
3997      train   324966  400000 2013-02-01              present   

                    Designation      JobCity Gender        DOB  10percentage  \
2              systems engineer      Chennai      f 1992-08-03         85.00   
11      senior systems engineer         Pune      m 1990-02-10         81.00   
15             network engineer           -1      f 1990-01-17         69.54   
21            software engineer           -1      f 1991-08-20         75.67   
29       java software engineer      Gurgaon      m 1992-06-11         64.00   
...                         ...          ...    ...        ...           ...   
3976  graduate engineer trainee  Navi Mumbai      f 1992-09-13         64.20   
3977          software engineer       Mumbai      m 1990-02-27         62.00   
3978          software engineer    Hyderabad      f 1991-04-24         89.70   
3993          software engineer   New Delhi       m 1987-04-15         52.09   
3997    senior systems engineer      Chennai      f 1991-02-26         70.60   

      ... ComputerScience  MechanicalEngg  ElectricalEngg TelecomEngg  \
2     ...              -1              -1              -1          -1   
11    ...              -1              -1              -1          -1   
15    ...              -1              -1              -1          -1   
21    ...              -1              -1              -1          -1   
29    ...             376              -1              -1          -1   
...   ...             ...             ...             ...         ...   
3976  ...             561              -1              -1          -1   
3977  ...              -1              -1              -1          -1   
3978  ...              -1              -1              -1          -1   
3993  ...              -1              -1              -1          -1   
3997  ...              -1              -1              -1          -1   

      CivilEngg  conscientiousness agreeableness  extraversion  nueroticism  \
2            -1             0.2718        1.7109        0.1637     -0.86820   
11           -1            -2.2720        1.1248        0.8171     -0.40780   
15           -1             1.7465        1.9048        0.6720     -0.29020   
21           -1            -1.0355       -0.5913       -1.3599      0.06223   
29           -1            -3.1994       -1.8393        0.0552      0.06230   
...         ...                ...           ...           ...          ...   
3976         -1             0.4155        0.0459        0.0100      0.27270   
3977         -1            -1.9629       -0.2793       -0.3440      0.53230   
3978         -1             0.2718        1.2114        0.4711     -0.10760   
3993         -1            -0.1082        0.3448        0.2366      0.64980   
3997         -1            -1.1128       -0.2793       -0.6343      1.32553   

      openess_to_experience  
2                    0.6721  
11                  -0.4455  
15                   1.2528  
21                  -1.3539  
29                  -0.7615  
...                     ...  
3976                 0.0973  
3977                -0.1295  
3978                 0.8637  
3993                -0.9194  
3997                -0.6035  

[660 rows x 38 columns]

## Observation: 
There are 660 Persons from INFORMATION TECHNOLOGY stream, where most of them are SOFTWARE & SYSTEM ENGINEERS.

In [37]:
new_df.get_group('instrumentation and control engineering')

Unnamed: 0       ID   Salary        DOJ                  DOL  \
14        train  1279958   300000 2013-07-01              present   
68        train    86373   150000 2010-07-01  2012-01-01 00:00:00   
569       train   604399   445000 2013-08-01              present   
690       train   609356   440000 2013-10-01  2015-04-01 00:00:00   
693       train   785663   220000 2014-09-01              present   
758       train   275243   300000 2011-08-01              present   
771       train   134963  1300000 2011-03-01              present   
1011      train  1154787   335000 2014-03-01  2015-04-01 00:00:00   
1117      train    60599   265000 2010-12-01              present   
1140      train  1015209   300000 2014-11-01              present   
1222      train   850679   335000 2013-12-01              present   
1797      train  1260265   300000 2012-10-01  2015-04-01 00:00:00   
1964      train  1015635   320000 2015-03-01              present   
2211      train   609211   700000 2014-01-01              present   
2342      train   518336   400000 2013-09-01  2015-04-01 00:00:00   
2509      train   716400   625000 2013-08-01              present   
2744      train   847661   440000 2014-01-01              present   
3114      train  1081371   305000 2014-05-01  2015-03-01 00:00:00   
3400      train   612330   200000 2014-05-01  2015-03-01 00:00:00   
3900      train   677954   200000 2014-02-01              present   

                      Designation       JobCity Gender        DOB  \
14         java software engineer     Bangalore      m 1992-07-02   
68        recruitment coordinator       Chennai      f 1989-03-18   
569             software engineer       Gurgaon      f 1990-10-22   
690            software developer       Gurgaon      f 1989-12-02   
693    service and sales engineer     New Delhi      m 1991-08-01   
758                  dcs engineer        Mumbai      m 1990-04-28   
771                junior manager         angul      m 1986-05-12   
1011           programmer analyst       Chennai      f 1992-09-15   
1117          technical assistant       Kolkata      f 1987-09-06   
1140              quality analyst       Gurgaon      f 1992-09-13   
1222           programmer analyst     Bangalore      m 1992-04-02   
1797                     lecturer  Gandhinagar       m 1990-11-01   
1964    associate system engineer     Ahmedabad      m 1992-05-02   
2211  business technology analyst     Bangalore      f 1991-07-17   
2342      business office manager         Unnao      m 1986-01-04   
2509  business technology analyst       Gurgaon      f 1991-11-28   
2744            software engineer         Noida      m 1991-06-18   
3114                 it recruiter   BHUBANESWAR      m 1992-12-23   
3400             product engineer     Faridabad      m 1990-12-24   
3900            quality assurance            -1      f 1992-06-07   

      10percentage  ... ComputerScience  MechanicalEngg  ElectricalEngg  \
14           81.20  ...             376              -1              -1   
68           78.00  ...              -1              -1              -1   
569          87.80  ...              -1              -1              -1   
690          83.16  ...              -1              -1              -1   
693          73.60  ...              -1              -1              -1   
758          67.20  ...              -1              -1              -1   
771          72.50  ...              -1              -1              -1   
1011         86.30  ...              -1              -1              -1   
1117         79.37  ...              -1              -1              -1   
1140         92.00  ...              -1              -1              -1   
1222         90.00  ...              -1              -1              -1   
1797         92.00  ...              -1              -1              -1   
1964         82.60  ...              -1              -1              -1   
2211         92.20  ...              -1              -1

## Observation: 
There are 20 Persons from INSTRUMENTATION AND CONTROL ENGINEERING stream, where most of them are SOFTWARE & PRODUCT ENGINEERS.

In [38]:
new_df = df.groupby('Gender')
new_df.first()

Unnamed: 0      ID  Salary        DOJ      DOL  \
Gender                                                  
f           train  203097  420000 2012-06-01  present   
m           train  579905  500000 2013-09-01  present   

                    Designation    JobCity        DOB  10percentage  \
Gender                                                                
f       senior quality engineer  Bangalore 1990-02-19          84.3   
m             assistant manager     Indore 1989-10-04          85.4   

                               10board  ...  ComputerScience  MechanicalEngg  \
Gender                                  ...                                    
f       board ofsecondary education,ap  ...               -1              -1   
m                                 cbse  ...               -1              -1   

       ElectricalEngg  TelecomEngg  CivilEngg conscientiousness agreeableness  \
Gender                                                                          
f                  -1           -1         -1            0.9737        0.8128   
m                  -1           -1         -1           -0.7335        0.3789   

        extraversion  nueroticism  openess_to_experience  
Gender                                                    
f             0.5269       1.3549                -0.4455  
m             1.2396      -0.1076                 0.8637  

[2 rows x 38 columns]

In [39]:
new_df.groups.keys()

dict_keys(['f', 'm'])

## MEAN: It is used to calculate mean/average of a given list of numbers. It returns mean of the data set passed as parameters. 

In [40]:
new_df['collegeGPA'].mean()

Gender
f    74.048056
m    70.679947
Name: collegeGPA, dtype: float64

In [41]:
new_df['Salary'].mean()

Gender
f    294937.304075
m    311716.211772
Name: Salary, dtype: float64

## Observation:
1. Females have Salary near to 2.9Lakhs with the COLLEGE-GPA of 74.
2. Males have Salary near to 3.1Lakhs with the COLLEGE-GPA of 74.

In [42]:
new_df.Salary.describe()

count           mean            std      min       25%       50%  \
Gender                                                                      
f        957.0  294937.304075  191031.789503  35000.0  200000.0  300000.0   
m       3041.0  311716.211772  218999.537438  35000.0  180000.0  300000.0   

             75%        max  
Gender                       
f       355000.0  3500000.0  
m       380000.0  4000000.0

In [43]:
df = df[['Gender', 'Specialization', 'ComputerProgramming', 'ElectronicsAndSemicon', 'ComputerScience',  'MechanicalEngg', 
        'ElectricalEngg', 'TelecomEngg', 'CivilEngg']]
print(type(df))
df

<class 'pandas.core.frame.DataFrame'>


Gender                             Specialization  ComputerProgramming  \
0         f                       computer engineering                  445   
1         m  electronics and communication engineering                   -1   
2         f                     information technology                  395   
3         m                       computer engineering                  615   
4         m  electronics and communication engineering                   -1   
...     ...                                        ...                  ...   
3993      m                     information technology                  345   
3994      f  electronics and communication engineering                  325   
3995      m                       computer engineering                  405   
3996      f             computer science & engineering                  445   
3997      f                     information technology                  435   

      ElectronicsAndSemicon  ComputerScience  MechanicalEngg  ElectricalEngg  \
0                        -1               -1              -1              -1   
1                       466               -1              -1              -1   
2                        -1               -1              -1              -1   
3                        -1               -1              -1              -1   
4                       233               -1              -1              -1   
...                     ...              ...             ...             ...   
3993                     -1               -1              -1              -1   
3994                    420               -1              -1              -1   
3995                     -1               -1              -1              -1   
3996                     -1              438              -1              -1   
3997                     -1               -1              -1              -1   

      TelecomEngg  CivilEngg  
0              -1         -1  
1              -1         -1  
2              -1         -1  
3              -1         -1  
4              -1         -1  
...           ...        ...  
3993           -1         -1  
3994           -1         -1  
3995           -1         -1  
3996           -1         -1  
3997           -1         -1  

[3998 rows x 9 columns]

## FINDING THE DATA OF MALES

In [44]:
df.loc[df.Gender == 'm', : ]

Gender                             Specialization  ComputerProgramming  \
1         m  electronics and communication engineering                   -1   
3         m                       computer engineering                  615   
4         m  electronics and communication engineering                   -1   
5         m  electronics and communication engineering                  645   
6         m             computer science & engineering                  405   
...     ...                                        ...                  ...   
3989      m             computer science & engineering                  465   
3991      m           electronics & telecommunications                  455   
3992      m                          civil engineering                   -1   
3993      m                     information technology                  345   
3995      m                       computer engineering                  405   

      ElectronicsAndSemicon  ComputerScience  MechanicalEngg  ElectricalEngg  \
1                       466               -1              -1              -1   
3                        -1               -1              -1              -1   
4                       233               -1              -1              -1   
5                        -1              407              -1              -1   
6                        -1              346              -1              -1   
...                     ...              ...             ...             ...   
3989                     -1              346              -1              -1   
3991                    400               -1              -1              -1   
3992                     -1               -1              -1              -1   
3993                     -1               -1              -1              -1   
3995                     -1               -1              -1              -1   

      TelecomEngg  CivilEngg  
1              -1         -1  
3              -1         -1  
4              -1         -1  
5              -1         -1  
6              -1         -1  
...           ...        ...  
3989           -1         -1  
3991          393         -1  
3992           -1        480  
3993           -1         -1  
3995           -1         -1  

[3041 rows x 9 columns]

## Observation:
There are 3041 Males who are from various EDUCATIONAL BACKGROUND.

## FINDING THE DATA OF FEMALES

In [45]:
df.loc[df.Gender == 'f', : ]

Gender                             Specialization  ComputerProgramming  \
0         f                       computer engineering                  445   
2         f                     information technology                  395   
15        f                     information technology                  355   
19        f                       computer engineering                  525   
20        f                       computer engineering                  445   
...     ...                                        ...                  ...   
3988      f            information science engineering                  445   
3990      f  electronics and communication engineering                  285   
3994      f  electronics and communication engineering                  325   
3996      f             computer science & engineering                  445   
3997      f                     information technology                  435   

      ElectronicsAndSemicon  ComputerScience  MechanicalEngg  ElectricalEngg  \
0                        -1               -1              -1              -1   
2                        -1               -1              -1              -1   
15                       -1               -1              -1              -1   
19                       -1               -1              -1              -1   
20                       -1               -1              -1              -1   
...                     ...              ...             ...             ...   
3988                     -1               -1              -1              -1   
3990                    400               -1              -1              -1   
3994                    420               -1              -1              -1   
3996                     -1              438              -1              -1   
3997                     -1               -1              -1              -1   

      TelecomEngg  CivilEngg  
0              -1         -1  
2              -1         -1  
15             -1         -1  
19             -1         -1  
20             -1         -1  
...           ...        ...  
3988           -1         -1  
3990          233         -1  
3994           -1         -1  
3996           -1         -1  
3997           -1         -1  

[957 rows x 9 columns]

## Observation:
There are 957 Females who are from various EDUCATIONAL BACKGROUND.

##  set_index( ) function: It is used to set the DataFrame index using existing columns. Set the DataFrame index (row labels) using one or more existing columns or arrays of the correct length. The index can replace the existing index or expand on it.

In [46]:
df.set_index('Gender', inplace = True)
df.head()

Specialization  ComputerProgramming  \
Gender                                                                   
f                            computer engineering                  445   
m       electronics and communication engineering                   -1   
f                          information technology                  395   
m                            computer engineering                  615   
m       electronics and communication engineering                   -1   

        ElectronicsAndSemicon  ComputerScience  MechanicalEngg  \
Gender                                                           
f                          -1               -1              -1   
m                         466               -1              -1   
f                          -1               -1              -1   
m                          -1               -1              -1   
m                         233               -1              -1   

        ElectricalEngg  TelecomEngg  CivilEngg  
Gender                                          
f                   -1           -1         -1  
m                   -1           -1         -1  
f                   -1           -1         -1  
m                   -1           -1         -1  
m                   -1           -1         -1

## iloc[  ] function: It returns a Pandas Series when one row is selected, and a Pandas DataFrame when multiple rows are selected, or if any column in full is selected. To counter this, pass a single-valued list if you require DataFrame output.

In [48]:
df.iloc[2]

Specialization           information technology
ComputerProgramming                         395
ElectronicsAndSemicon                        -1
ComputerScience                              -1
MechanicalEngg                               -1
ElectricalEngg                               -1
TelecomEngg                                  -1
CivilEngg                                    -1
Name: f, dtype: object

In [49]:
df.iloc[3000]

Specialization           electronics & telecommunications
ComputerProgramming                                   555
ElectronicsAndSemicon                                 292
ComputerScience                                        -1
MechanicalEngg                                         -1
ElectricalEngg                                         -1
TelecomEngg                                            -1
CivilEngg                                              -1
Name: f, dtype: object

In [50]:
df.iloc[3990]

Specialization           electronics and communication engineering
ComputerProgramming                                            285
ElectronicsAndSemicon                                          400
ComputerScience                                                 -1
MechanicalEngg                                                  -1
ElectricalEngg                                                  -1
TelecomEngg                                                    233
CivilEngg                                                       -1
Name: f, dtype: object

In [51]:
df.iloc[1, 2]

466

In [52]:
df.iloc[7, 7]

-1

In [55]:
df.iloc[55, 7]

-1

In [56]:
df.iloc[390, :]

Specialization           information technology
ComputerProgramming                         445
ElectronicsAndSemicon                        -1
ComputerScience                             346
MechanicalEngg                               -1
ElectricalEngg                               -1
TelecomEngg                                  -1
CivilEngg                                    -1
Name: f, dtype: object

In [57]:
df.iloc[3, :]

Specialization           computer engineering
ComputerProgramming                       615
ElectronicsAndSemicon                      -1
ComputerScience                            -1
MechanicalEngg                             -1
ElectricalEngg                             -1
TelecomEngg                                -1
CivilEngg                                  -1
Name: m, dtype: object

## SELECTING MULTIPLE ROWS

In [58]:
df.iloc[[2, 3, 6]]

Specialization  ComputerProgramming  \
Gender                                                        
f               information technology                  395   
m                 computer engineering                  615   
m       computer science & engineering                  405   

        ElectronicsAndSemicon  ComputerScience  MechanicalEngg  \
Gender                                                           
f                          -1               -1              -1   
m                          -1               -1              -1   
m                          -1              346              -1   

        ElectricalEngg  TelecomEngg  CivilEngg  
Gender                                          
f                   -1           -1         -1  
m                   -1           -1         -1  
m                   -1           -1         -1

In [59]:
df.iloc[[2, 3, 6],[1, 2, 3]]

ComputerProgramming  ElectronicsAndSemicon  ComputerScience
Gender                                                             
f                       395                     -1               -1
m                       615                     -1               -1
m                       405                     -1              346

In [60]:
df.iloc[[2, 3, 6], :]

Specialization  ComputerProgramming  \
Gender                                                        
f               information technology                  395   
m                 computer engineering                  615   
m       computer science & engineering                  405   

        ElectronicsAndSemicon  ComputerScience  MechanicalEngg  \
Gender                                                           
f                          -1               -1              -1   
m                          -1               -1              -1   
m                          -1              346              -1   

        ElectricalEngg  TelecomEngg  CivilEngg  
Gender                                          
f                   -1           -1         -1  
m                   -1           -1         -1  
m                   -1           -1         -1

In [61]:
df.iloc[3:5]

Specialization  ComputerProgramming  \
Gender                                                                   
m                            computer engineering                  615   
m       electronics and communication engineering                   -1   

        ElectronicsAndSemicon  ComputerScience  MechanicalEngg  \
Gender                                                           
m                          -1               -1              -1   
m                         233               -1              -1   

        ElectricalEngg  TelecomEngg  CivilEngg  
Gender                                          
m                   -1           -1         -1  
m                   -1           -1         -1

In [62]:
df.iloc[:, 4]

Gender
f   -1
m   -1
f   -1
m   -1
m   -1
    ..
m   -1
f   -1
m   -1
f   -1
f   -1
Name: MechanicalEngg, Length: 3998, dtype: int64

In [63]:
df.iloc[:, 2:4]

ElectronicsAndSemicon  ComputerScience
Gender                                        
f                          -1               -1
m                         466               -1
f                          -1               -1
m                          -1               -1
m                         233               -1
...                       ...              ...
m                          -1               -1
f                         420               -1
m                          -1               -1
f                          -1              438
f                          -1               -1

[3998 rows x 2 columns]

In [64]:
df.iloc[555]

Specialization           biotechnology
ComputerProgramming                355
ElectronicsAndSemicon              300
ComputerScience                     -1
MechanicalEngg                      -1
ElectricalEngg                      -1
TelecomEngg                         -1
CivilEngg                           -1
Name: m, dtype: object

## FINAL CONCLUSION

## OBSERVATION FROM THE DATA:

1. 76.06% (3041) of Males & 23.94% (957) are working at different LOCATIONS with various SPECIALIZATION & DESIGNATIONS.
2. Males have high Salary of 40 Lakhs.
3. 880 Persons are of ELECTRONICS AND COMMUNICATION ENGINEERING & 660 are of INFORMATION TECHNOLOGY Streams.
4. 3701 Persons are from COLLEGE-TIER - 2.
5. 8 Persons have COLLGE-GPA > 80.
6. Most of them are DESIGNATED as SOFTWARE ENGINEERS.